.github/workflows/trivy.yml

```diff
@@ -20,62 +20,48 @@ env:
   TRIVY_JAVA_DB_REPOSITORY: public.ecr.aws/aquasecurity/trivy-java-db
 permissions:
   contents: read
+  security-events: write  # for github/codeql-action/upload-sarif to upload SARIF results
```

Comment:
Move the write permissions under each specific job that requires it - do not make it workflow-global.

Reply:
done